server/oauth2.go

```diff
@@ -405,6 +407,19 @@ func (s *Server) parseAuthorizationRequest(r *http.Request) (req storage.AuthReq
 		return req, &authErr{"", "", errServerError, ""}
 	}
 
+	if connectorID != "" {
+		connectors, err := s.storage.ListConnectors()
+		if err != nil {
+			return req, &authErr{"", "", errServerError, "Unable to retrieve connectors"}
+		}
+		for _, c := range connectors {
+			if c.ID == connectorID {
+				break
+			}
+			return req, &authErr{"", "", errInvalidRequest, "Invalid ConnectorID"}
```

Comment:
With this _inside the `for` loop_, it'll fail if there are more than one connectors, and the second one is the requested one. It would be nice if we had another test case for this.

Reply:
Added a validate function.   Added a second connector to the test server.  Added test.

Reply:
All right.So I ended up duplicating newTestServer to have newTestServerMultipleConnectors.  It seems that the code workflow tests may depend on having a single connector, so adding in a second connector in order to test broke all of that.

Reply:
This now tests for selecting a connector other than the first.